src/Config.AzureKeyVault/AzureKeyVaultConfigurationExtensions.cs

```diff
@@ -124,6 +125,38 @@ public static class AzureKeyVaultConfigurationExtensions
             return result.AccessToken;
         }
 
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Azure KeyVault.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="vault">Azure KeyVault uri.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAzureKeyVault(
+            this IConfigurationBuilder configurationBuilder,
+            string vault)
+        {
+            return AddAzureKeyVault(configurationBuilder, vault, null);
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Azure KeyVault.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="vault">Azure KeyVault uri.</param>
+        /// <param name="manager">The <see cref="IKeyVaultSecretManager"/> instance used to control secret loading.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddAzureKeyVault(
+            this IConfigurationBuilder configurationBuilder,
+            string vault,
+            IKeyVaultSecretManager manager)
+        {
+            var azureServiceTokenProvider = new AzureServiceTokenProvider();
+            var authenticationCallback = new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback);
+            var keyVaultClient = new KeyVaultClient(authenticationCallback);
+            
+            return AddAzureKeyVault(configurationBuilder, vault, keyVaultClient, manager);
```

Comment:
Why are there no null checks on manager all the way through the stack? It looks like it will null ref here: https://github.com/aspnet/Configuration/blob/2ba3a05af48c83f3471abc771d9fb03f39e09000/src/Config.AzureKeyVault/AzureKeyVaultConfigurationProvider.cs#L52

Reply:
You could also go the other way and just use the DefaultKeyVaultSecretManager when no manager is specified... assuming that is the sensible thing to do